### PR TITLE
Fix small grammar mistakes in chapter 2

### PR DESCRIPTION
**Pull request by icholy**
These are the mistakes I encountered in chapter two.

### CODE REVIEW

**Comment by icholy** [comment]
@pedropark99 comment "revert" on any changes you don't like and I'll remove them.

File: Chapters/03-structs.qmd [hunk 1]
```diff
@@ -726,10 +726,10 @@ While the `deinit()` method is the method used for destroying an existing instan
 
 The `init()` and `deinit()` methods are both used extensively in Zig code, and you will see both of
 them being used when we talk about allocators in @sec-allocators.
-But, as another example, let's build a simple `User` struct to represent an user of some sort of system.
+But, as another example, let's build a simple `User` struct to represent a user of some sort of system.
```
Comment (pedropark99) [inline Chapters/03-structs.qmd:729]: In theory, the use of "an" here is correct, because the next word ("user") begins with a consonant sound.

**Review by pedropark99 — CHANGES_REQUESTED** [review]
Just revert the change made in the use of "an" (last comment), and we are good to merge this PR 🥳.

File: Chapters/03-structs.qmd (hunk 1 above)
Comment (icholy) [reply Chapters/03-structs.qmd:729]: https://english.stackexchange.com/a/105117

Comment (icholy) [reply Chapters/03-structs.qmd:729]: Let me know if you still want me to revert.

Comment (pedropark99) [reply Chapters/03-structs.qmd:729]: Uhmm very interesting, I didn't knew this rule. In this case, let's then merge as is 😉

**Review by pedropark99 — APPROVED** [review]

**Comment by pedropark99** [comment]
Thank you once again for the fixes 😉!

**Comment by icholy** [comment]
Thanks for writing this book :)

**Pull request merged**